src/Main.purs

```diff
@@ -262,9 +265,9 @@ main = void $ runAff failed succeeded do
 runWithArgs :: Args.Args -> AffN Unit
 runWithArgs args = do
   out <- getOutputter args
-  _ <- validate out
+  ver <- validate out
   watch <- getFlag "watch" args.globalOpts
-  args' <- addProject args
```

Comment:
Instead of having code that's specific to `pulp init` in here, how about using `Pulp.Validate.getPursVersion` inside `Pulp.Init`?

Reply:
Yes, I'm more concerned about the factoring of the code. I would be surprised if running `purs --version` twice was noticeable, but if you're concerned about that, I think a good option could be to cache the first result and return that on subsequent executions of that command?

Reply:
Okay, I didn't actually test it. Just did now, it seems indeed to be something that you wouldn't notice, especially next to the dependency downloads and builds. I'll just move the code to `Init` and we can cache it later if this ever becomes a problem.